agent.go

```diff
@@ -305,6 +309,53 @@ func (s *sandbox) teardownSharedPidNs() error {
 	return nil
 }
 
+func (s *sandbox) listenToUdevEvents() {
+	fieldLogger := agentLog.WithField("subsystem", "udevlistener")
+
+	uEvHandler, err := uevent.NewHandler()
+	if err != nil {
+		fieldLogger.Warnf("Error starting uevent listening loop %s", err)
+		return
+	}
+	defer uEvHandler.Close()
+
+	for {
+		uEv, err := uEvHandler.Read()
+		if err != nil {
+			fieldLogger.Error(err)
+			continue
+		}
+
+		fieldLogger = fieldLogger.WithFields(logrus.Fields{
+			"uevent-action":    uEv.Action,
+			"uevent-devpath":   uEv.DevPath,
+			"uevent-subsystem": uEv.SubSystem,
+			"uevent-seqnum":    uEv.SeqNum,
+			"uevent-devname":   uEv.DevName,
+		})
+
+		// Check if device hotplug event results in a device node being created.
+		if uEv.DevName != "" && uEv.Action == "add" && strings.HasPrefix(uEv.DevPath, rootBusPath) {
+			s.Lock()
```

Comment:
You could add a quick comment explaining the lock is needed to safely read and modify the `pciDeviceMap` map.